doc/cephfs/nfs.rst

```diff
@@ -79,3 +79,219 @@ Current limitations
+
+        [root@minikube /]# ceph orchestrator mon update 3
+        Updated mon count to 3
+
+It takes few minutes for the monitor daemon to come up. Check the status with::
+
+        [root@minikube /]# ceph orchestrator service ls
+        mgr a minikube rook-ceph-mgr-a-55964c5888-w9sbt None None
+        mon a minikube rook-ceph-mon-a-579c967b98-dv7tw None None
+        mon b minikube rook-ceph-mon-b-784b5b5db5-dpnpf None None
+        mon c minikube rook-ceph-mon-c-585d8d5f6b-25v2l None None
+        osd 0 minikube rook-ceph-osd-0-b9666466-r7678 None None
+
+Create a Ceph File System
+-------------------------
+Using ceph-mgr volumes module, we will create a ceph file system::
+
+        [root@minikube /]# ceph fs volume create myfs
+
+By default replicated size for OSD is 3. Since we are using only one OSD. It can cause error. Let's fix this up by setting replicated size to 1.::
+
+        [root@minikube /]# ceph osd pool set cephfs.myfs.meta size 1
+        [root@minikube /]# ceph osd pool set cephfs.myfs.data size 1
+
+Check Cluster status again::
+
+        [root@minikube /]# ceph -s
+          cluster:
+            id:     3a30f44c-a9ce-4c26-9f25-cc6fd23128d0
+            health: HEALTH_OK
+
+          services:
+            mon: 3 daemons, quorum a,b,c (age 27m)
+            mgr: a(active, since 27m)
+            mds: myfs:1 {0=myfs-a=up:active} 1 up:standby-replay
+            osd: 1 osds: 1 up (since 56m), 1 in (since 56m)
+
+          data:
+            pools:   2 pools, 24 pgs
+            objects: 22 objects, 2.2 KiB
+            usage:   5.1 GiB used, 11 GiB / 16 GiB avail
+            pgs:     24 active+clean
+
+          io:
+            client:   639 B/s rd, 1 op/s rd, 0 op/s wr
+
+Create a NFS-Ganesha Server Cluster
+-----------------------------------
+Add Storage for NFS-Ganesha Servers to prevent recovery conflicts::
+
+        [root@minikube /]# ceph osd pool create nfs-ganesha 64
+        pool 'nfs-ganesha' created
+        [root@minikube /]# ceph osd pool set nfs-ganesha size 1
+        [root@minikube /]# ceph orchestrator nfs add mynfs nfs-ganesha mynfs
+
+Scale out NFS-Ganesha cluster::
+
+        [root@minikube /]# ceph orchestrator nfs update mynfs 2
+
+Configure NFS-Ganesha Exports
+-----------------------------
+Initially rook creates ClusterIP service for the dashboard. With this service
+type, only the pods in same kubernetes cluster can access it.
+
+Expose Ceph Dashboard port::
+
+        kubectl patch service -n rook-ceph -p '{"spec":{"type": "NodePort"}}' rook-ceph-mgr-dashboard
+        kubectl get service -n rook-ceph rook-ceph-mgr-dashboard
+        NAME                      TYPE       CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
+        rook-ceph-mgr-dashboard   NodePort   10.108.183.148   <none>        8443:31727/TCP   117m
+
+This makes the dashboard reachable outside kubernetes cluster and the service
+type is changed to NodePort service.
+
+Create JSON file for dashboard::
+
```

Comment:
I'd suggest wrapping this file in a cat command using a heredoc. That way, a reader can cut and paste straight out of the doc.